fitbenchmarking/core/tests/test_fitting_benchmarking.py

```diff
@@ -0,0 +1,824 @@
+ROOT = os.getcwd()
+DATA_DIR = ROOT + \
+    "/fitbenchmarking/benchmark_problems/NIST/average_difficulty/"
+
+
+def mock_loop_over_hessians_func_call(controller):
+    """
+    Mock function for the __loop_over_hessian method
+    """
+    result_args = {'controller': controller,
+                   'accuracy': 1,
+                   'runtimes': [2],
+                   'emissions': 3,
+                   'runtime_metric': 'mean'}
+    result = FittingResult(**result_args)
+    return [result]
+
+
+def mock_loop_over_jacobians_func_call(controller):
+    """
+    Mock function for the __loop_over_jacobians method
+    """
+    controller.cost_func.jacobian = Analytic(controller.cost_func.problem)
+    result = mock_loop_over_hessians_func_call(controller)
+    return result
+
+
+def mock_loop_over_minimizers_func_call(controller, minimizers):
+    """
+    Mock function for the __loop_over_minimizers method
+    """
+    results = []
+    for _ in minimizers:
+        result = mock_loop_over_jacobians_func_call(controller)
+        results.extend(result)
+    return results, []
+
+
+def mock_loop_over_softwares_func_call(cost_func):
+    """
+    Mock function for the __loop_over_softwares method
+    """
+    controller = ScipyController(cost_func)
+    controller.cost_func.jacobian = Analytic(controller.cost_func.problem)
+    controller.parameter_set = 0
+    results = []
+    for _ in range(9):
+        result = mock_loop_over_hessians_func_call(controller)
+        results.extend(result)
+    return results
+
+
+def mock_loop_over_cost_func_call(problem):
+    """
+    Mock function for the __loop_over_cost_func method
+    """
+    cost_func = WeightedNLLSCostFunc(problem)
+    results = mock_loop_over_softwares_func_call(cost_func)
+    return results
+
+
+def mock_loop_over_cost_func_call_all_fail(problem):
+    """
+    Mock function for the __loop_over_cost_func method
+    """
+    cost_func = WeightedNLLSCostFunc(problem)
+    controller = ScipyController(cost_func)
+    controller.cost_func.jacobian = Analytic(controller.cost_func.problem)
+    controller.parameter_set = 0
+    result = FittingResult(**{'controller': controller,
+                              'accuracy': np.inf,
+                              'runtimes': [2],
+                              'emissions': 3,
+                              'runtime_metric': 'mean'})
+    return [result]
+
+
+def mock_loop_over_starting_values(problem):
+    """
+    Mock function for the __loop_over_starting_values method
+    """
+    cost_func = WeightedNLLSCostFunc(problem)
+    controller = ScipyController(cost_func)
+    controller.cost_func.jacobian = Analytic(controller.cost_func.problem)
+    controller.parameter_set = 0
+    result = mock_loop_over_hessians_func_call(controller)
+    return result
+
+
+def set_up_controller(file, options):
+    """
+    Sets up the controller for the PerformFitTests
+    """
+    data_file = DATA_DIR + file
```

Comment:
```suggestion
    data_file = os.path.join(DATA_DIR, file)
```

And similarly for all other cases that use DATA_DIR + .. .